### PR TITLE
Add React Native demo app to CI workflow

### PR DESCRIPTION
**Pull request by aryaemami59**
This PR:

- [X] Adds React Native example app to CI workflow.

### CODE REVIEW

**Comment by markerikson** [comment]
We need both of these to actually do some kind of a real build step, so that we know that things at least compile.  Can you update them to do an actual build instead of `echo "Done"`?

**Comment by aryaemami59** [comment]
> We need both of these to actually do some kind of a real build step, so that we know that things at least compile. Can you update them to do an actual build instead of `echo "Done"`?

Yes I will work on that, tbh I just had to take a break from it since RN is not my favorite thing to work with :)

**Comment by aryaemami59** [comment]
I will update this as soon as Reselect 5.0.2 with the new `createSelector.withTypes<RootState>()` API is released.
I will probably have to submit another PR to do the same for all the other examples.

**Comment by markerikson** [comment]
Any idea why it errored with `gradlew not found`?

**Comment by aryaemami59** [comment]
@markerikson I think so, I'll see if I can fix it.

File: examples/publish-ci/react-native/src/constants/TypedColors.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+  primary: string
+  white: string
+  lighter: string
+  light: string
+  dark: string
+  darker: string
+  black: string
+}
+
+export const TypedColors = Colors as AllColors
```
Comment (EskiMojo14) [inline examples/publish-ci/react-native/src/constants/TypedColors.ts:13]: out of curiosity, why not just add a declaration for the original Colors variable?

Comment (aryaemami59) [reply examples/publish-ci/react-native/src/constants/TypedColors.ts:13]: Oh I just remembered, it wasn't missing a declaration, the `Colors` variable was just typed as `any`. And this was my way of overriding it to make it more type safe. Is there a better a way to accomplish that without doing what I did there?

Comment (EskiMojo14) [reply examples/publish-ci/react-native/src/constants/TypedColors.ts:13]: you're already providing types for `"react-native/Libraries/NewAppScreen"` in `globals.d.ts`, just add the Colors export to it 

```diff
declare module "react-native/Libraries/NewAppScreen" {
  import type { FC } from "react"
  export const HermesBadge: FC
+
+  export const Colors: {
+    primary: string;
+    white: string;
+    lighter: string;
+    light: string;
+    dark: string;
+    darker: string;
+    black: string;
+  };
}
```

Comment (aryaemami59) [reply examples/publish-ci/react-native/src/constants/TypedColors.ts:13]: @EskiMojo14 I get `Cannot redeclare block-scoped variable 'Colors'.ts(2451) LaunchScreen.d.ts(15, 16): 'Colors' was also declared here.`

Comment (EskiMojo14) [reply examples/publish-ci/react-native/src/constants/TypedColors.ts:13]: ah, interesting - apparently my dependencies hadn't actually installed 

then yeah, i guess this is necessary - shame the original types are so rubbish

**Pull request merged**